src/functional/iterator/builtins.py

```diff
@@ -136,3 +163,141 @@ def make_tuple(*args):
 @builtin_dispatch
 def tuple_get(*args):
     raise BackendNotSelectedError()
+
+
+# FIXME(ben): various of the following built-ins will shadow python built-ins!
+# We should find a good way around this...
+
+@builtin_dispatch
+def abs(*args):
+    raise BackendNotSelectedError()
+
+
+@builtin_dispatch
+def min(*args):
+    raise BackendNotSelectedError()
+
+
+@builtin_dispatch
+def max(*args):
+    raise BackendNotSelectedError()
+
+
+@builtin_dispatch
+def mod(*args):
+    raise BackendNotSelectedError()
+
+
+@builtin_dispatch
+def sin (*args):
+    raise BackendNotSelectedError()
+
+
+@builtin_dispatch
+def cos(*args):
+    raise BackendNotSelectedError()
+
+
+@builtin_dispatch
+def tan(*args):
+    raise BackendNotSelectedError()
+
+
+@builtin_dispatch
+def arcsin(*args):
+    raise BackendNotSelectedError()
+
+
+@builtin_dispatch
+def arccos(*args):
+    raise BackendNotSelectedError()
+
+
+@builtin_dispatch
+def arctan(*args):
+    raise BackendNotSelectedError()
+
+
+@builtin_dispatch
+def sinh(*args):
+    raise BackendNotSelectedError()
+
+
+@builtin_dispatch
+def cosh(*args):
+    raise BackendNotSelectedError()
+
+
+@builtin_dispatch
+def tanh(*args):
+    raise BackendNotSelectedError()
+
+
+@builtin_dispatch
+def arcsinh(*args):
+    raise BackendNotSelectedError()
+
+
+@builtin_dispatch
+def arccosh(*args):
+    raise BackendNotSelectedError()
+
+
+@builtin_dispatch
+def arctanh(*args):
+    raise BackendNotSelectedError()
+
+
+@builtin_dispatch
+def sqrt(*args):
+    raise BackendNotSelectedError()
+
+
+@builtin_dispatch
+def exp(*args):
+    raise BackendNotSelectedError()
+
+
+@builtin_dispatch
+def log(*args):
+    raise BackendNotSelectedError()
+
+
+@builtin_dispatch
+def gamma(*args):
+    raise BackendNotSelectedError()
+
+
+@builtin_dispatch
+def cbrt(*args):
+    raise BackendNotSelectedError()
+
+
+@builtin_dispatch
+def isfinite(*args):
+    raise BackendNotSelectedError()
+
+
+@builtin_dispatch
+def isinf(*args):
+    raise BackendNotSelectedError()
+
+
+@builtin_dispatch
+def isnan(*args):
+    raise BackendNotSelectedError()
+
+
+@builtin_dispatch
+def floor(*args):
+    raise BackendNotSelectedError()
+
+
+@builtin_dispatch
+def ceil(*args):
+    raise BackendNotSelectedError()
+
+
+@builtin_dispatch
+def trunc(*args):
+    raise BackendNotSelectedError()
```

Comment:
I would suggest to have a seperate list (of strings) for the builtins that can be appended to `__all__` and in addition be used to automatically create all of these functions (e.g. using `globals()[builtin_name] = builtin_dispatch(stub)`. This has the disadvantage that it might break some IDEs introspection, but at least we avoid the code bloat. Your call.